ansible/roles/create-s390x-vm/tasks/main.yml

```diff
@@ -0,0 +1,74 @@
+- name: Fetch the variables from var file
+  include_vars:
+    file: vars.yml
+
+- name: Check for existing VSI
+  ibm.cloudcollection.ibm_is_instance_info:
+    name: "{{ vsi_name }}"
+  failed_when:
+    - vsi.rc != 0
+    - '"No Instance found" not in vsi.stderr'
+  register: vsi
+
+- name: Configure VSI
+  ibm.cloudcollection.ibm_is_instance:
+    name: "{{ vsi_name }}"
+    state: available
+    id: "{{ vsi.resource.id | default(omit) }}"
+    vpc: "{{ vpc_id }}"
+    profile: "{{ vsi_profile }}"
+    image: "{{ vsi_image_id }}"
+    resource_group: "{{ vsi_resource_group_id }}"
+    keys:
+      - "{{ ssh_key_id }}"
+    primary_network_interface:
+     - subnet: "{{ subnet_id }}"
+    zone: "{{ zone }}"
+  register: vsi
+
+- name: Check for existing Floating IP
+  ibm.cloudcollection.ibm_is_floating_ip_info:
+    name: "{{ vsi_name }}-fip"
+  failed_when:
+    - fip.rc != 0
+    - '"No floatingIP found" not in fip.stderr'
+  register: fip
+
+- name: Configure Floating IP Address
+  ibm.cloudcollection.ibm_is_floating_ip:
+    name: "{{ vsi_name }}-fip"
+    state: available
+    id: "{{ fip.resource.id | default(omit) }}"
+    resource_group: "{{ vsi_resource_group_id }}"
+    target: "{{ vsi.resource.primary_network_interface[0]['id'] }}"
+  register: fip
+
+- name: Print Floating IP Address
+  debug:
+    msg: "{{ vsi_name }} IP Address: {{ fip.resource.address }}"
+
+- name: Add VSI to Ansible inventory
+  add_host:
+    name: "{{ fip.resource.address }}"
+    ansible_user: root
+    groups:
+      - job_id_{{ job_id }}
+      - platform_{{ test_platform }}
+    ansible_ssh_extra_args: "-o StrictHostKeyChecking=no"
+    ansible_ssh_private_key_file: "~/.ssh/acs-sshkey_rsa.prv"
+    vm_config: "{{ vm_config }}"
+    vm_image_family: "{{ vm_image_family }}"
+    vm_collection_method: "{{ vm_collection_method }}"
+
+- name: Register server with Red Hat
```

Comment:
This feels like a provisioning step rather than a creation step

Reply:
I think I heard Robby mentioned the x86 vms are created (through gcp) with subscription attached. And i don't see there is a registration process for x86 vm creation/provision. So I consider the registration one part of the vm creation process.

Reply:
For the purposes of these roles and playbooks, the distinction elsewhere is whether or not a connection to the VM is established for the step to take place. Anything that involves configuring the VM should be performed under the provisioning role, and I think that registering the subscription fits within that.

Reply:
sure, I have followed the suggestion and made the subscription under provisioning role.